packages/stream/index.js

```diff
@@ -20,6 +20,11 @@
  * }} PromiseKit
  */
 
+// TypeScript ReadOnly semantics are not sufficiently expressive to distinguish
+// a value one promises not to alter from a value one must not alter,
+// making it useless.
```

Comment:
I'm not sure I understand or agree with this statement. What is the harm in marking the result as `readonly`? Besides the fact that a lot of code is currently typed to only accept a non-readonly object.

Reply:
Freezing the IterationResult objects makes them Readonly and they cannot be passed where IterationResult is expected, because the receiver hasn’t marked them as “WontTouch” (const& in C++ parlance), even though the receiver does not touch.

If TypeScript were useful for communicating Readonly, const reference marking would have to be pervasive. C++ does this and the cognitive burden on the developer is tremendous.

Reply:
Yeah so the problem is that most code, including official typings, are not strict enough in their readonly-ness. There might be an argument to help fix some of those. But I agree in the mean time, if it blows up too many things not in out control, then that's fine.

However I don't understand how `Readonly<IteratorResult<T>>` is different than `IteratorResult`. I guess declaring as an interface doesn't mean the methods are readonly, which is weird but understandable.